classes/image_types_ostree.bbclass

```diff
@@ -129,7 +129,7 @@ IMAGE_CMD_ostree () {
 IMAGE_TYPEDEP_ostreecommit = "ostree"
 do_image_ostreecommit[depends] += "ostree-native:do_populate_sysroot"
 do_image_ostreecommit[lockfiles] += "${OSTREE_REPO}/ostree.lock"
-IMAGE_CMD_ostreecommit () {
+IMAGE_CMD:ostreecommit () {
```

Comment:
I'm a bit confused about the syntax changes here. Is this right that the `IMAGE_CMD` bits need colons, but `IMAGE_TYPEDEP` doesn't? Seems inconsistent to me. Anyway, these aren't overrides but rather task names... unless the underlying mechanism is somehow the same.

Reply:
Currently this is correct as IMAGE_TYPEDEP is used like this:
```
meta/classes/image.bbclass:        deps = (d.getVar('IMAGE_TYPEDEP_' + t) or "").split()
meta/classes/image.bbclass:        vardeps.add('IMAGE_TYPEDEP_' + t)
```
while IMAGE_CMD was changed to:
```
meta/classes/image.bbclass:        vardepsexclude = (d.getVarFlag('IMAGE_CMD:' + realt, 'vardepsexclude', True) or '').split()
meta/classes/image.bbclass:        image_cmd = localdata.getVar("IMAGE_CMD")
meta/classes/image.bbclass:        vardeps.add('IMAGE_CMD:' + realt)
meta/classes/image.bbclass:            bb.fatal("No IMAGE_CMD defined for IMAGE_FSTYPES entry '%s' - possibly invalid type name or missing support class" % t)
meta/classes/image.bbclass:        # Since a copy of IMAGE_CMD:xxx will be inlined within do_image_xxx,
meta/classes/image.bbclass:        # prevent a redundant copy of IMAGE_CMD:xxx being emitted as a function.
meta/classes/image.bbclass:        d.delVarFlag('IMAGE_CMD:' + realt, 'func')
```

But I've asked Richard if it should be changed in oe-core as well to avoid this confusion.

Reply:
Richard changed the IMAGE_TYPEDEP and CONVERSION_CMD to use : as well, it was added to oe-core/master yesterday and I've added 4th commit here to do the same.